js/data.js

```diff
@@ -166,7 +166,7 @@ class Data extends AdaptCollection {
   }
 
   async triggerDataReady(newLanguage) {
-    if (newLanguage) {
+    if (newLanguage && newLanguage !== Adapt.config.get('_activeLanguage')) {
```

Comment:
```suggestion
    const hasLanguageChanged = (newLanguage !== Adapt.config.get('_activeLanguage'));
    if (newLanguage && hasLanguageChanged) {
```

Reply:
Thanks Oliver. I think I will deleting this draft PR though after all. The language comparison at this point is no good to us as they will always be equal